samples/apps/copilot-chat-app/SKWebApi/Program.cs

```diff
@@ -24,28 +25,42 @@ public static void Main(string[] args)
             serverPort = SKWebApiConstants.DefaultServerPort;
         }
 
+        // Set the protocol to use
         string protocol = "https";
+        bool useHttp = builder.Configuration.GetSection("UseHttp").Get<bool>();
+        if (useHttp)
+        {
+            protocol = "http";
+        }
+
         builder.WebHost.UseUrls($"{protocol}://*:{serverPort}");
 
         // Add services to the DI container
         AddServices(builder.Services, builder.Configuration);
 
         var app = builder.Build();
 
+        var logger = app.Services.GetRequiredService<ILogger>();
+
         // Configure the HTTP request pipeline
         if (app.Environment.IsDevelopment())
         {
             app.UseSwagger();
             app.UseSwaggerUI();
         }
         app.UseCors();
-        app.UseHttpsRedirection();
         app.UseAuthorization();
         app.MapControllers();
 
         // Log the health probe URL
-        var logger = app.Services.GetService<ILogger>()!;
-        logger.LogInformation("Health probe: https://{0}:{1}/probe", protocol, serverPort);
+        string hostName = Dns.GetHostName();
+        logger.LogInformation("Health probe: {Protocol}://{Host}:{Port}/probe", protocol, hostName, serverPort);
+
+        if (useHttp)
+        {
+            logger.LogWarning("Server is using HTTP instead of HTTPS. Do not use HTTP in production.\n" +
```

Comment:
[nit] Please remove  '\n' and '!' from log messages. \n doesn't scale outside of the console and "!" can be considered jarring or rude in some cultures.

Reply:
I guess I'm just too thick-skinned! :D  Removing those.